source/org/zfin/jbrowse/presentation/JBrowseImageBuilder.java

```diff
@@ -70,6 +75,39 @@ public GenomeBrowserImage build() {
         return new JBrowseImage(this);
     }
 
+    private String getHighlightLandmarkByMarkerOrFeature(ZdbID entity) {
+        String defaultValue = "";
+        if (entity instanceof Marker) {
+            defaultValue = ((Marker) entity).getAbbreviation();
+        } else if (entity instanceof Feature) {
+            defaultValue = ((Feature) entity).getAbbreviation();
+        }
+        String landmark = defaultValue;
+
+        GenomeLocation location = getGenomeLocation(entity);
+
+        if (location != null) {
+            landmark = location.getChromosome() + ":" +
+                    location.getStart() + ".." +
+                    location.getEnd();
+        }
+        return landmark;
+    }
+
+    private GenomeLocation getGenomeLocation(ZdbID markerOrFeature) {
+        List<GenomeLocation> genomeLocations = getLinkageRepository().getGenericGenomeLocation(markerOrFeature);
+
+        Optional<GenomeLocation> first = genomeLocations.stream().filter(
+                genomeLocation -> getGenomeBuild().getValue().equals(genomeLocation.getAssembly())
+        ).findFirst();
+
+        GenomeLocation result = null;
```

Comment:
Better to use the exception safe orElse() method instead of get(), e.g.
return first.orElse(null);